seqr/views/apis/staff_api_tests.py

```diff
@@ -0,0 +1,173 @@
+
+TEST_INDEX_SRC_PATH="test_index_src_path"
+TEST_INDEX_SECOND_SRC_PATH="test_index_second_src_path"
+TEST_INDEX_OLD_SRC_PATH="test_index_old_src_path"
+TEST_INDEX_1KG_SRC_PATH="test_index_1kg_src_path"
+ES_INDEX_MAPPING={
+  "test_index": {
+    "mappings": {
+      "variant": {
+        "_meta": {
+          "gencodeVersion": "25",
+          "genomeVersion": "38",
+          "sampleType_1": "WES",
+          "sourceFilePath_1": "gs://seqr-datasets/v02/GRCh38/RDG_WES_Broad_Internal/v13/RDG_Broad_WES_Internal.vcf.bgz",
+          "genomeVersion_1": "38",
+          "sampleType": "WES",
+          "sourceFilePath": TEST_INDEX_SRC_PATH,
+          "gencodeVersion_1": "25"
+        },
+        "_all": {
+          "enabled": False
+        }
+      }
+    }
+  },
+  "test_index_second": {
+    "mappings": {
+      "variant": {
+        "_meta": {
+          "gencodeVersion": "25",
+          "hail_version": "0.2.24",
+          "genomeVersion": "38",
+          "sampleType_1": "WGS",
+          "sourceFilePath_1": "gs://seqr-datasets/v02/GRCh38/RDG_WGS_Broad_Internal/v10/sharded_vcf/RDG_Broad_WGS_Internal.filtered.*.vcf.gz",
+          "genomeVersion_1": "38",
+          "sampleType": "WGS",
+          "sourceFilePath": TEST_INDEX_SECOND_SRC_PATH,
+          "hail_version_1": "0.2.24",
+          "gencodeVersion_1": "25"
+        },
+        "_all": {
+          "enabled": False
+        },
+      }
+    }
+  },
+  "test_index_old": {
+    "mappings": {
+      "variant": {
+        "_meta": {
+          "gencodeVersion": "25",
+          "hail_version": "0.2.24",
+          "genomeVersion": "38",
+          "sampleType_1": "WGS",
+          "sourceFilePath_1": "gs://seqr-datasets/v02/GRCh38/RDG_WGS_Broad_Internal/v10/sharded_vcf/RDG_Broad_WGS_Internal.filtered.*.vcf.gz",
+          "genomeVersion_1": "38",
+          "sampleType": "WGS",
+          "sourceFilePath": TEST_INDEX_SRC_PATH,
+          "hail_version_1": "0.2.24",
+          "gencodeVersion_1": "25"
+        },
+        "_all": {
+          "enabled": False
+        },
+      }
+    }
+  },
+  "1kg.vcf.gz": {
+    "mappings": {
+      "variant": {
+        "_meta": {
+          "gencodeVersion": "19",
+          "genomeVersion": "37",
+          "sampleType": "WES",
+          "datasetType": "VARIANTS",
+          "sourceFilePath": TEST_INDEX_1KG_SRC_PATH
+        },
+        "_all": {
+          "enabled": False
+        },
+      }
+    }
+  }
+}
+from django.test import TestCase
+from django.urls.base import reverse
+
+from seqr.views.apis.staff_api import elasticsearch_status
+from seqr.views.utils.test_utils import _check_login
+
+class VariantSearchAPITest(TestCase):
+    fixtures = ['users', '1kg_project']
+    multi_db = True
+
+    @mock.patch('elasticsearch_dsl.index.Index.get_mapping')
+    @mock.patch('elasticsearch.client.cat.CatClient.aliases')
+    @mock.patch('elasticsearch.client.cat.CatClient.indices')
+    @mock.patch('elasticsearch.client.cat.CatClient.allocation')
```

Comment:
In general, you should never patch 3 functions on the same class separately. Just patch the class once, and then set the return value for each functionon that same mock.

However, in this case you have gone too deeply into the class in your patching. You should really only patch libraries that are imported and used directly in the file being tested, never dive into the libraries themselves. seqr doesn't care how the internals of ibraries work and should not be testing them. I would either patch the helper function `get_es_client` which is what is imported here, or if you wanted you could patch the one library call within that function and patch `elasticsearch.Elasticsearch`. Either way, it would be a single patch for all the cat calls and then you would set the return values for the different cat calls by doing something like 
```
mock_client.cat.allocation.return_value = ES_CAT_ALLOCATION
mock_client.cat.indices.return_value = ES_CAT_INDICES
...
```

Reply:
Cool, I will try.

Reply:
Since the function under test doesn't use helper functions, so I have to mock the library classes of methods.